### DISCUSSION TITLE
how to allow some compartments to have negative concentrations

### DISCUSSION THREAD

**Discussion by hainiwen** [Open-Systems-Pharmacology/Forum#1588] [Questions & Problems] [answered]
![Capture](https://github.com/Open-Systems-Pharmacology/Forum/assets/56790121/daff4b02-f58e-4828-9510-3e2bfe6bd15a)
Hi there!
I have some error here.
While I am pretty sure that the mass transfer should be right, as the concentrations are quite low, I cannot compute the current model with a Tolerance about 1E-9
When I set it as Tol 1E-4, it works but the concentration curves are quite bumpy when it goes below 1E-5.

So I was wondering, can I keep the Tolerance, but allow negative values in some compartments. I heard it's do-able, but I don't know where is the setting...

Thank you in advance.

Haini

**Comment by hainiwen** [comment]
Or allow variables to be negative?

**Comment by Yuri05** [comment, marked as answer]
You can change the setting for each output in the Molecule Start Values building block and then update your simulation.
After that, you should plot all the curves which were identified as having negative values before and check if the negative values are acceptable (can be considered as numerical noise) or not.
If the values are not acceptable: either there is something wrong with the model or the solver settings must be adjusted.
![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/ac238d42-919e-400c-898d-996212dfa22e)

**Reply by hainiwen** [reply]
it worked! thanks!
